t/unit-tests/u-hashmap.c

```diff
@@ -165,39 +163,26 @@ static void t_add(struct hashmap *map, unsigned int ignore_case)
 
 		hashmap_for_each_entry_from(map, entry, ent)
 		{
-			int ret;
-			if (!check_int((ret = key_val_contains(
-						key_val, seen,
-						ARRAY_SIZE(key_val), entry)),
-				       ==, 0)) {
-				switch (ret) {
-				case 1:
-					test_msg("found entry was not given in the input\n"
-						 "    key: %s\n  value: %s",
-						 entry->key, get_value(entry));
-					break;
-				case 2:
-					test_msg("duplicate entry detected\n"
-						 "    key: %s\n  value: %s",
-						 entry->key, get_value(entry));
-					break;
-				}
+			int ret = key_val_contains(key_val, seen,
+							ARRAY_SIZE(key_val), entry);
+			cl_assert(ret == 0);
+			if (ret == 1) {
+				cl_check_fail("found entry was not given in the input\n");
+			} else if (ret == 2) {
+				cl_check_fail("duplicate entry detected\n");
```

Comment:
This if/else-cascade is impossible to reach: we already use `cl_assert(ret == 0)`, which would cause us to stop executing the testcase when the assert is violated. We can probably just drop it.